micro_sam/evaluation/benchmark_datasets.py

```diff
@@ -0,0 +1,499 @@
+        image_crops, gt_crops = [], []
+        for i, (per_n_instance, per_id) in enumerate(sorted(zip(n_instances, n_ids), reverse=True), start=1):
+            crop_box = crop_boxes[per_id]
+            crop_image, crop_gt = image[crop_box], gt[crop_box]
+            # NOTE: We avoid using the crops which do not match the desired tile shape.
+            if skip_smaller_shape and crop_image.shape != tile_shape:
+                continue
+
+            # NOTE: There could be a case where some later patches are invalid.
+            if per_n_instance == 1:
+                break
+
+            image_crops.append(crop_image)
+            gt_crops.append(crop_gt)
+
+            # NOTE: If the number of patches extracted have been fulfiled, we stop sampling patches.
+            if len(image_crops) > 0 and i >= crops_per_input:
+                break
+
+        # Finally, let's store all the patches
+        for image_crop, gt_crop in zip(image_crops, gt_crops):
+            fname = f"{dataset_choice}_{image_counter:05}.tif"
+            assert image_crop.shape == gt_crop.shape
+            imageio.imwrite(os.path.join(save_image_dir, fname), image_crop, compression="zlib")
+            imageio.imwrite(os.path.join(save_gt_dir, fname), gt_crop, compression="zlib")
+            image_counter += 1
+
+    return save_image_dir, save_gt_dir, ndim
+
+
+def _get_image_label_paths(image_dir, gt_dir):
+    image_paths = natsorted(glob(os.path.join(image_dir, "*")))
+    gt_paths = natsorted(glob(os.path.join(gt_dir, "*")))
+    return image_paths, gt_paths
+
+
+def _run_automatic_segmentation_per_dataset(
+    image_paths: List[Union[os.PathLike, str]],
+    gt_paths: List[Union[os.PathLike, str]],
+    model_type: str,
+    output_folder: Union[os.PathLike, str],
+    device: Optional[Union[torch.device, str]] = None,
+    ndim: Optional[int] = None,
+    checkpoint_path: Union[os.PathLike, str] = None,
+    **auto_seg_kwargs
+):
+    """Functionality to run automatic segmentation for multiple input files at once.
+    It stores the evaluated automatic segmentation results (quantitative).
+
+    Args:
+        image_paths: List of filepaths for the input image data.
+        gt_paths: List of filepaths for the corresponding label data.
+        model_type: The choice of model type for Segment Anything.
+        output_folder: Filepath to the folder where we store all the results.
+        ndim: The number of input dimensions.
+        checkpoint_path: The filepath for stored checkpoints.
+        auto_seg_kwargs: Additional arguments for automatic segmentation parameters.
+    """
+    result_path = os.path.join(output_folder, "results", "automatic_segmentation.csv")
+    prediction_dir = os.path.join(output_folder, "automatic_instance_segmentation", "inference")
+    os.makedirs(prediction_dir, exist_ok=True)
+
+    for image_path in tqdm(image_paths, desc="Run automatic segmentation"):
```

Comment:
This is currently a bit ineffecient: we load the model again for each image.I think for running AIS that can be quite some overhead.

Reply:
Not sure how exactly we want to address it. One option would be enabling passing multiple paths to `automatic_instance_segmentation`.

Reply:
I think I'll update `automatic_instance_segmentation` so that it can consider multiple paths. I'll take care of it. Thanks for the suggestion!

Reply:
It was slightly bit more complicated to enable multiple paths in the existing function, so I refactored a few things to avoid doing redundant things (eg. calling the predictor and segmenter for every image). This part should be taken care of now.